src/login/logind-dbus.c

```diff
@@ -2232,6 +2257,14 @@ static int method_schedule_shutdown(sd_bus_message *message, void *userdata, sd_
         if (r != 0)
                 return r;
 
+        if (!timestamp_is_set(elapse) && m->maintenance_window) {
```

Comment:
as mentioned I think this should just check for elapse == UINT64_MAX.

Also, this should generate a clean error if no maintainance window is configured.

Reply:
by this I mean, please add a new proper dbus error in src/libsystemd/sd-bus/bus-common-errors.[ch], which you return here.